Split out transactions server actions into separate files

nswai983 opened the pull request:
Moving transactions server actions into separate files per issue [#1113](https://github.com/actualbudget/actual/issues/1113).

No functional changes in this PR.

FYI this is my first contribution to actual, so please let me know if I miss anything!

ACWalker commented on packages/loot-core/src/server/transactions/app.ts:
```diff
@@ -0,0 +1,112 @@
+  'transactions-batch-update',
+  mutator(async function ({ added, deleted, updated, learnCategories }) {
+    return withUndo(async () => {
+      const result = await batchUpdateTransactions({
```
Think you can just return this directly

ACWalker replied:
Can just return directly here.

ACWalker commented on packages/loot-core/src/server/transactions/app.ts:
```diff
@@ -0,0 +1,112 @@
+app.method(
+  'transactions-batch-update',
+  mutator(async function ({ added, deleted, updated, learnCategories }) {
+    return withUndo(async () => {
```
```suggestion
    return withUndo(async () => {
      result = await batchUpdateTransactions({
        added,
        updated,
        deleted,
        learnCategories,
      });
    });
```

nswai983 commented:
@ACWalker there were some merge conflicts I addressed that kept the PR from merging. I addressed those and your prior comment about returning directly in the 'transactions-batch-update' handler.

nswai983 commented:
Ran into some issues with Partials as a result of some changes to main. I have addressed these issues, and the current commit is up to date and passes all checks.

matt-fidd reviewed (approved):
Thanks for this!

nswai983 commented on packages/loot-core/src/server/transactions/types/handlers.ts:
```diff
@@ -0,0 +1,39 @@
+// @ts-strict-ignore
```
Ah I hadn't checked the types here. I ran into issues with the types being 'any' in lint checks so I used the types I did based on context surrounding the code being used.

I can update the types to match the packages/loot-core/src/types/server-handlers.d.ts file.

nswai983 replied:
Question: Should the non-typed params (transaction, options, etc.) remain untyped? Or should types be added to these? I used TransactionEntity for the transaction paramaters to add some clarity surrounding them.

jfdoming replied:
Makes sense. I think adding extra types is fine! As long as we can be sure they're the right types. Feel free to also leave off improving the types to a separate PR, these things can be pretty in depth

matt-fidd commented:
Hey @nswai983 - how are you doing on marrying up the types between this and `packages/loot-core/src/types/server-handlers.d.ts`?
